include/deal.II/multigrid/mg_constrained_dofs.h

```diff
@@ -311,60 +312,54 @@ MGConstrainedDoFs::initialize(
 
       // Loop through relevant cells and faces finding those which are periodic
       // neighbors.
-      typename DoFHandler<dim, spacedim>::cell_iterator cell = dof.begin(l),
-                                                        endc = dof.end(l);
-      for (; cell != endc; ++cell)
-        if (cell->level_subdomain_id() != numbers::artificial_subdomain_id)
-          {
-            for (auto f : cell->face_indices())
-              if (cell->has_periodic_neighbor(f) &&
-                  cell->periodic_neighbor(f)->level() == cell->level())
-                {
-                  if (cell->is_locally_owned_on_level())
-                    {
-                      Assert(
-                        cell->periodic_neighbor(f)->level_subdomain_id() !=
-                          numbers::artificial_subdomain_id,
-                        ExcMessage(
-                          "Periodic neighbor of a locally owned cell must either be owned or ghost."));
-                    }
-                  // Cell is a level-ghost and its neighbor is a
-                  // level-artificial cell nothing to do here
-                  else if (cell->periodic_neighbor(f)->level_subdomain_id() ==
-                           numbers::artificial_subdomain_id)
+      for (const auto &cell : dof.active_cell_iterators_on_level(l))
```

Comment:
Shouldn't this be
```suggestion
      for (const auto &cell : dof.cell_iterators_on_level(l))
```
Some tests fail (in fact, I am surprised that not more are failing), and I believe this is the reason.